### PR TITLE
Introduce moving funds timeout reset delay

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: #252

The `resetMovingFundsTimeout` function can be called multiple times in a row if the live wallets count is zero. This is dangerous since that function will be reimbursable and a malicious maintainer can drain the reimbursement pool. To protect against that, we introduce a reset delay that will allow resetting the timeout just before it is about to be exceeded.

### CODE REVIEW

File: solidity/contracts/bridge/BridgeState.sol [hunk 1]
```diff
@@ -91,6 +91,13 @@ library BridgeState {
         // than `redemptionDustThreshold` in order to prevent redemption requests
         // with values lower or equal to `movingFundsDustThreshold`.
         uint64 movingFundsDustThreshold;
+        // Time after which the moving funds timeout can be reset in case the
+        // target wallet commitment cannot be submitted due to a lack of live
+        // wallets in the system. It is counted from the moment when the wallet
+        // was requested to move their funds and switched to the MovingFunds
+        // state. Value in seconds. This value should be smaller than the value
```
Comment (pdyraga) [inline solidity/contracts/bridge/BridgeState.sol:98]:
```suggestion
        // state. Value in seconds. This value should be lower than the value
```

See [this](https://github.com/keep-network/tbtc-v2/pull/241#discussion_r870114337) conversation please.

File: solidity/test/bridge/Bridge.MovingFunds.test.ts [hunk 2]
```diff
@@ -715,33 +716,69 @@ describe("Bridge - Moving funds", () => {
...
+                .to.emit(bridge, "MovingFundsTimeoutReset")
+                .withArgs(ecdsaWalletTestData.pubKeyHash160)
+            })
           })
 
-          it("should emit MovingFundsTimeoutReset event", async () => {
-            await expect(tx)
-              .to.emit(bridge, "MovingFundsTimeoutReset")
-              .withArgs(ecdsaWalletTestData.pubKeyHash160)
+          context("when reset delay was not elapsed yet", () => {
```
Comment (pdyraga) [inline solidity/test/bridge/Bridge.MovingFunds.test.ts:758]:
```suggestion
          context("when reset delay has not elapsed yet", () => {
```

File: solidity/contracts/bridge/Bridge.sol [hunk 3]
```diff
@@ -1159,10 +1162,17 @@ contract Bridge is Governable, EcdsaWalletOwner, Initializable {
...
+    ///        wallets having their BTC balance below that value can begin
+    ///        closing immediately as transferring such a low value may not be
+    ///        possible due to BTC network fees.
+    /// @param movingFundsTimeoutResetDelay New value of the moving funds
+    ///        timeout reset delay in seconds. It is the time after which the
+    ///        moving funds timeout can be reset in case the target wallet
+    ///        commitment cannot be submitted due to a lack of live wallets
+    ///        in the system. It is counted from the moment when the wallet
+    ///        was requested to move their funds and switched to the MovingFunds
+    ///        state.
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply solidity/contracts/bridge/Bridge.sol]: See https://github.com/keep-network/tbtc-v2/pull/282/commits/90f362a87287334b6c65ebdcc7d04342827155e1

File: solidity/contracts/bridge/BridgeState.sol (hunk 1 above)
Comment (lukasz-zimnoch) [reply solidity/contracts/bridge/BridgeState.sol]: See https://github.com/keep-network/tbtc-v2/pull/282/commits/90f362a87287334b6c65ebdcc7d04342827155e1

File: solidity/test/bridge/Bridge.MovingFunds.test.ts (hunk 2 above)
Comment (lukasz-zimnoch) [reply solidity/test/bridge/Bridge.MovingFunds.test.ts]: See https://github.com/keep-network/tbtc-v2/pull/282/commits/90f362a87287334b6c65ebdcc7d04342827155e1

File: solidity/test/bridge/Bridge.MovingFunds.test.ts [hunk 4]
```diff
@@ -715,33 +716,69 @@ describe("Bridge - Moving funds", () => {
...
+              await restoreSnapshot()
+            })
+
+            it("should revert", async () => {
+              await expect(
+                bridge.resetMovingFundsTimeout(
+                  ecdsaWalletTestData.pubKeyHash160
+                )
+              ).to.be.revertedWith("Moving funds timeout cannot be reset yet")
+            })
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply solidity/test/bridge/Bridge.MovingFunds.test.ts:781]: See https://github.com/keep-network/tbtc-v2/pull/282/commits/1f13c309a90f17911b1e70255ac7674d0e1139ed

**Review by pdyraga — APPROVED** [review]

**Pull request merged**